pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTResource.java

```diff
@@ -20,6 +20,9 @@ public ASTResource(JavaParser p, int id) {
     public Object jjtAccept(JavaParserVisitor visitor, Object data) {
         return visitor.visit(this, data);
     }
+
+    // TODO Should we deprecate all methods from ASTFormalParameter?
```

Comment:
I mean override all methods that ASTResource inherits from ASTFormalParameter in ASTResource (just calling super), and deprecate the overrides. AFAIK that wouldn't deprecate the methods in ASTFormalParameter, just in ASTResource

Reply:
You mean, because ASTResource should not have inherited from ASTFormalParameter in the first place? (https://docs.oracle.com/javase/specs/jls/se11/html/jls-14.html#jls-14.20.3)

The method `isArray()` and `getArrayDepth()` are deprecated, even if used via `ASTResource` - but other methods of course are not deprecated.

We can tackle this in a later change/PR.